### DISCUSSION TITLE
The Mirage

### DISCUSSION THREAD

**Discussion by TheDeathlyCow** [TheDeathlyCow/scorchful#20] [Ideas]
I'm creating this discussion to record and organize my rough thoughts on how the Mirage boss will work in Scorchful, before coming up with specific tasks and issues. If anyone sees this and wants to leave a comment, feel free to do so. 

The Mirage will be a new mini? Boss mob (unsure if I will include a boss bar and everything) that spawns in the Desert/Badlands when the player is warm. It should be easier to see the warmer the player is. It will have a deserty look to it, and will serve as the basis for integrating Breezes into the natural world[^1]. The Mirage will float in the air so I don't have to make a walk-cycle animation. 

The current sequence for fighting the mob will look something like this (very rough outline):
1. The player reaches a temperature of 100% in the Desert, and receives the Heat Stroke effect.
2. The Heat Stroke effect causes the player to see 'desert mirages' of the Mirage (capital M)
3. These desert mirages should ONLY be visible to players with Heat Stroke.
4. Approaching a desert mirage will cause it to fade away in a cloud of dust. This cloud will be visible to all players. 
5. Approaching ~3 desert mirages within a few minutes will cause a Mirage to spawn on the last desert mirage. The Mirage is visible to all players. 
6. The player fights and possibly kills the Mirage. Like the Frostologer, heat should be a part of its attacks and fight. For example, having a higher heat level should make the Mirage easier to see. Some attack ideas: 
    - Fear/Mesmer as described below 
    - Spawning Breeze minions, especially when multiple players are nearby
    - Generally wind based attacks 
    - Spawning clones of itself (like the Illusioner does)
8. Killing the Mirage will drop a magical fibrous items. It could be based on Asebestos[^2] but could also just be a simple Scorched Silk item
9. The item can be used to craft a mask/visage that, when activated, allows a player to leave a "mirage" of themself behind. the actual player will be totally invisible while its active, including armour. 
    - the decoy mirage player will have no armour and significantly reduced max health 
    - if the decoy mirage is killed the player is snapped back and the Fear effect will be applied to the, 
10. The item can also be used to create something that functions as a thermometer but doesn't exactly look like it. 
    - for 26.2+: I will probably incorporate cinnabar into the crafting recipe, as cinnabar is actually a major source of Mercury in real life. but it wont be essential for 26.1. 

[^1]: I feel that Breezes being exclusive to an underground is a very weird design decision. Underground areas are precisely the opposite of breezy and airy. trial chambers also feel kind of "side questy" to me rather than a normal part of the world. 

[^2]: Asbestos has long been known as a "magical" fibrous material that cannot catch fire. It is certainly extremely toxic and dangerous to handle, which could be an element of the item. The only reason I would avoid asbestos is that its better known as a heat resistance/fire proofing material than it is as a temperature sensitive substance like mercury.

**Comment by TheDeathlyCow** [comment]
## Some ideas on the look of the Mirage

The Breeze added in 1.21 will provide the basis for the Mirage aesthetic. The Mirage will essentially be a larger "Desert" breeze. This will also allow me to integrate Breezes more into the natural world, as I feel that their exclusivity to an underground side-quest in vanilla is a bit odd.

The aesthetic will combine the breeze with the idea of Lady Midday/Noonwraiths from Slavic folklore/The Witcher 

<img width="150" height="181" alt="image" src="https://github.com/user-attachments/assets/7cea3f0b-9be0-4ac2-95a1-d660f9250ebf" />

https://en.wikipedia.org/wiki/Lady_Midday

https://brendan-noble.com/poludnitsa-poludnica-noonwraith-slavic-demon-of-noon-slavic-mythology-saturday/

<img width="250" height="365" alt="image" src="https://github.com/user-attachments/assets/4c5ff381-9858-4138-9108-41fe2cf87ece" />

<details>
<summary>old idea, may still pull elements from</summary>

It would have a very Egyptian-inspired appearance. Sobek, the Crocodile God of the Nile, could be a good basis for the look, as a crocodile dropping teeth feels like a fairly natural interaction. 

![302px-Sobek_mirror svg](https://github.com/TheDeathlyCow/scorchful/assets/53499406/10b81436-09a5-4a01-8856-af510a3cce01)
![sobek-egyptian-god-7](https://github.com/TheDeathlyCow/scorchful/assets/53499406/65532093-51cb-4352-bd11-5ca173350662)

A more ghostly appearance would also help justify the fact that it floats in the air instead of walking.

Desert Wraith art: https://www.artstation.com/artwork/n0vKEr

![6i15d8lbjds91](https://github.com/TheDeathlyCow/scorchful/assets/53499406/33d34b6a-46bd-4a19-b8e6-01d9f1040a21)

Spice Navigators from Dune should also be a good look for that floaty thing (though the 1984 version is a bit too gross for me tbh)

![atrfpb42o8t71](https://github.com/TheDeathlyCow/scorchful/assets/53499406/a2b5d9e1-a07d-42b8-81c5-440c1f9fb2c0)
![dune_guild_navigator_by_geniusfetus_dg27uk2-fullview](https://github.com/TheDeathlyCow/scorchful/assets/53499406/7e676cbb-1eaa-45d8-9dc1-cf6d5be00a6d)
</details>

**Comment by TheDeathlyCow** [comment]
# Deathstills 

The Fremen in Dune use Deathstills to collect water from the dead. Scorchful could add a Deathstill enchantment that can be applied to the stealth dagger of the Mirage. This enchantment would suck the water out of opponents when it kills them. This could also be an inherent ability of the dagger.

**Reply by Oroton** [reply]
It should be inherant to the dagger, but an enchantment would be cool And should be applied to the Trident.
gives trident extra use, but plays into the 'water' theme of the trident

**Reply by TheDeathlyCow** [reply]
im gonna move this feature to thirstful.

**Comment by TheDeathlyCow** [comment]
# Mesmer and Fear

An attack for the Mirage will be the status effect "Mesmerizing". When this effect is applied to the Mirage, it will check each tick to see if it can apply a "Mesmerized" effect to nearby entities. When the player or another entity is Mesmerized, they will look at and move slowly towards the closest Mesmerizing entity, and if they reach the Mesmerizing entity the effect will clear and the Mesmerized entity will take damage. 

When the Mesmerized effect is removed, the entity will be immediately inflicted with Fear. This will make them immune to being Mesmerized again for a short time, but with some drawbacks. Fear is currently being worked on in #72.

These effects will also be craftable as potions. Current thinking is that Warped Nether Lily petals/seeds/spores will brew the Mesmerizing Potion, and the Crimson petal/seed/spore will brew the Potion of Paranoia (which will grant both Fear and Darkness since they go well together).
